wp-content/themes/core/components/blocks/icon_grid/Icon_Grid_Controller.php

```diff
@@ -0,0 +1,251 @@
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_attrs(): string {
+		return Markup_Utils::concat_attrs( $this->attrs );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_container_classes(): string {
+		return Markup_Utils::class_attribute( $this->container_classes );
+	}
+
+	/**
+	 * @return string
+	 */
+	public function get_content_classes(): string {
+		$this->content_classes[] = 'g-3-up';
+		$this->content_classes[] = 'g-centered';
+
+		return Markup_Utils::class_attribute( $this->content_classes );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_header_args(): array {
+		if ( empty( $this->title ) && empty( $this->description ) ) {
+			return [];
+		}
+
+		return [
+			Content_Block_Controller::TAG     => 'header',
+			Content_Block_Controller::LAYOUT  => Content_Block_Controller::LAYOUT_CENTER,
+			Content_Block_Controller::LEADIN  => $this->get_leadin(),
+			Content_Block_Controller::TITLE   => $this->get_title(),
+			Content_Block_Controller::CONTENT => $this->get_content(),
+			Content_Block_Controller::CTA     => $this->get_cta(),
+			Content_Block_Controller::CLASSES => [
+				'c-block__content-block',
+				'c-block__header',
+				'b-icon-grid__header',
+			],
+		];
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_leadin(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::CLASSES => [
+				'c-block__leadin',
+				'b-icon-grid__leadin',
+				'h6',
+			],
+			Text_Controller::CONTENT => $this->leadin ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_title(): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::TAG     => 'h2',
+			Text_Controller::CLASSES => [
+				'c-block__title',
+				'h3',
+			],
+			Text_Controller::CONTENT => $this->title ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	private function get_content(): Deferred_Component {
+		return defer_template_part( 'components/container/container', null, [
+			Container_Controller::CLASSES => [
+				'c-block__description',
+				'b-icon-grid__description',
+				't-sink',
+				's-sink',
+			],
+			Container_Controller::CONTENT => $this->description ?? '',
+		] );
+	}
+
+	/**
+	 * @return Deferred_Component
+	 */
+	public function get_cta(): Deferred_Component {
+		$cta = wp_parse_args( $this->cta, [
+			'content' => '',
+			'url'     => '',
+			'target'  => '',
+		] );
+
+		return defer_template_part( 'components/link/link', null, [
+			Link_Controller::URL     => $cta['url'],
+			Link_Controller::CONTENT => $cta['content'] ?: $cta['url'],
+			Link_Controller::TARGET  => $cta['target'],
+			Link_Controller::CLASSES => [
+				'c-block__cta-link',
+				'a-btn',
+				'a-btn--has-icon-after', 
+				'icon-arrow-right'
+			],
+		] );
+	}
+
+	public function get_icon_card_args() {
+		if ( empty( $this->icons ) ) {
+			return [];
+		}
+
+		$cards = [];
```

Comment:
Can move this above the empty check and then just `return $cards;` if `$this->icons` is empty.